src/nnvm/low_precision_pass.cc

```diff
@@ -29,374 +29,336 @@
-                          new_node);
-            }
-          } else {
-            if (mirror_fp32_map.count(node_entry)) {
-              new_node->inputs.emplace_back(mirror_fp32_map[node_entry]);
-            } else {
-              ObjectPtr mirror_node  = mirror_map.at(node_entry.node.get());
-              NodeEntry mirror_entry = NodeEntry{mirror_node, node_entry.index, node_entry.version};
-              std::string suffix     = GetSuffix(node_entry, mirror_map);
-              AddCastNode(node_entry, suffix, mirror_entry, "float32", &mirror_fp32_map, new_node);
-            }
-          }
-        }
-      }
-    } else if (CheckConditionalFP32(conditional_fp32_ops, excluded_syms, node)) {
-      for (size_t i = 0; i < node->num_outputs(); ++i) {
-        const auto out_entry       = NodeEntry(node, i, 0);
-        mirror_fp32_map[out_entry] = NodeEntry(new_node, i, 0);
+    node_map.emplace(old_node.get(), std::move(new_node));
+  });
+  for (const NodeEntry& old_out_ne : src.outputs) {
+    register_node_entry(old_out_ne, nullptr, nullptr);
+  }
+
+  // convert the model
+  DFSVisit(src.outputs, [&](const ObjectPtr& old_node) {
+    if (old_node->is_variable() || old_node->op() == Op::Get("amp_multicast") ||
+        IsCastOp(old_node->op())) {
+      const ObjectPtr& new_node = node_map.at(old_node.get());
+      for (const auto& old_ne : old_node->inputs) {
+        const ObjectPtr& new_in_node = node_map.at(old_ne.node.get());
+        new_node->inputs.emplace_back(new_in_node, old_ne.index, old_ne.version);
       }
-      for (size_t i = 0; i < node->inputs.size(); ++i) {
-        const auto& node_entry = node->inputs[i];
-        if (mirror_fp32_map.count(node_entry)) {
-          new_node->inputs.emplace_back(mirror_fp32_map[node_entry]);
-        } else if (std::find(mutable_inputs.begin(), mutable_inputs.end(), i) !=
-                   mutable_inputs.end()) {
-          // Can't insert amp_cast for this inputs. Such op have to handle fp32 inputs itself.
-          ObjectPtr mirror_node = mirror_map.at(node_entry.node.get());
-          new_node->inputs.emplace_back(mirror_node, node_entry.index, node_entry.version);
-        } else {
-          ObjectPtr mirror_node  = mirror_map.at(node_entry.node.get());
-          NodeEntry mirror_entry = NodeEntry{mirror_node, node_entry.index, node_entry.version};
-          std::string suffix     = GetSuffix(node_entry, mirror_map);
-          AddCastNode(node_entry, suffix, mirror_entry, "float32", &mirror_fp32_map, new_node);
-        }
+      return;
+    }
+
+    if (fp32_ops.count(old_node->op()->name) > 0 || excluded_syms.count(old_node->attrs.name) > 0) {
+      KeepOriginalNode(old_node, node_map, &entry_map);
+    } else if (target_dtype_ops.count(old_node->op()->name) > 0) {
+      if (!TryLowPrecision(target_dtype, old_node, node_map, nodes_entries, &entry_map)) {
+        LOG(WARNING) << "Conversion to low precision of a node: " + old_node->attrs.name +
```

Comment:
in the other places where TryLowPrecision is used there is ni warning message, is it intentional. If no, we can put the log within the TryLowPrecision function

Reply:
This is intentional. `TryLowPrecision` is also used below, where the failure is not something unexpected.